walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,79 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
-public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+public final class FindMeetingQuery {   
+
+    private static ArrayList <TimeRange>  findAvailableTimes(Collection<Event> events, Collection<String> mandatoryAttendees,long duration) {
+      ArrayList<TimeRange> availableTimeForAll = new ArrayList<TimeRange>();
+      int end = TimeRange.END_OF_DAY;
+      int start = TimeRange.START_OF_DAY;
+
+      if(mandatoryAttendees.isEmpty()){
+          availableTimeForAll.add(TimeRange.WHOLE_DAY);
+          return availableTimeForAll;
+      }
+    
+      if (duration > 60 * 24) {
+        return availableTimeForAll;
+      }
+
+      Set<String> attendeesSet = new HashSet<String>();
+      for(String attendee : mandatoryAttendees) {
+        attendeesSet.add(attendee);
+      }
+    
+      if(events.isEmpty()) {
+        availableTimeForAll.add(TimeRange.WHOLE_DAY);
+        return availableTimeForAll;
+      }
+     
+      for(Event event : events) {
+        Collection<String> eventAttendees = event.getAttendees();
```

Comment:
It might be more readable if we filter the events first in a separate function:

```
   Collection<Event> filterEventsByAttendees(Collection<Event> events, Set<String> attendees) {
   /* Returns events that have attendees in the attendees set. */
   }
```